R/merge.R

```diff
@@ -98,5 +100,15 @@ merge.data.table = function(x, y, by = NULL, by.x = NULL, by.y = NULL, all = FAL
 
   # retain custom classes of first argument that resulted in dispatch to this method, #1378
   setattr(dt, "class", class_x)
-  dt
+  if(info) {
+    message("Rows in x: " , x[, .N, verbose = F],"\n",
+   "Rows in y: " , y[, .N, verbose = F],"\n",
+    "Matched rows: " , y[x, nomatch = NULL, on = by, allow.cartesian =
+                              allow.cartesian, verbose = F][, .N, verbose = F],"\n",
+    "Total rows after merge: " , dt[, .N, verbose = F])
```

Comment:
```nrows(x)```, ```nrows(y)``` and ```nrows(dt)``` would be simpler than using ```.N```. However, on the join line, my limited testing indicates you could simply put ```j = .N``` instead of chaining another call.